src/store/__tests__/RelayFragmentResolver-test.js

```diff
@@ -0,0 +1,251 @@
+/**
+ * Copyright 2013-2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails oncall+relay
+ */
+
+'use strict';
+
+jest.dontMock('GraphQLStoreChangeEmitter');
+
+const Relay = require('Relay');
+const RelayFragmentResolver = require('RelayFragmentResolver');
+const RelayStoreData = require('RelayStoreData');
+const RelayTestUtils = require('RelayTestUtils');
+
+const readRelayQueryData = require('readRelayQueryData');
+
+describe('RelayFragmentResolver', () => {
+  let fragment;
+  let results;
+  let store;
+  let storeData;
+  let writer;
+
+  // helper functions
+  const {getNode, writePayload} = RelayTestUtils;
+
+  function genMockSubscriber() {
+    const onCompleted = jest.genMockFunction();
+    const onError = jest.genMockFunction();
+    const onNext = jest.genMockFunction();
+    const mockClear = () => {
+      [onCompleted, onError, onNext].forEach(fn => fn.mockClear());
+    };
+    return {
+      onCompleted,
+      onError,
+      onNext,
+      mockClear,
+    };
+  }
+
+  function createFragmentResolver(dataID) {
+    return new RelayFragmentResolver(
+      storeData,
+      fragment,
+      dataID
+    );
+  }
+
+  beforeEach(() => {
+    jest.resetModuleRegistry();
+    const concreteFragment = Relay.QL`fragment on Node{id,address{country}}`;
+    const query = getNode(Relay.QL`
+      query {
+        node(id: "123") {
+          ${concreteFragment},
+        }
+      }
+    `);
+    results = {
+      __dataID__: '123',
+      id: '123',
+      address: {
+        __dataID__: 'client:1',
+        country: 'US',
+      },
+    };
+    fragment = getNode(concreteFragment);
+    storeData = new RelayStoreData();
+    storeData.initializeGarbageCollector();
+    store = storeData.getRecordStore();
+    writer = storeData.getRecordWriter();
+    writePayload(store, writer, query, {node: results});
+
+    jest.addMatchers(RelayTestUtils.matchers);
+  });
+
+  describe('subscribe()', () => {
+    it('subscription `dispose` is idempotent', () => {
+      const resolver = createFragmentResolver('123');
+      const subscriber = genMockSubscriber();
+      const subscription = resolver.subscribe(subscriber);
+      subscription.dispose();
+      subscription.dispose();
```

Comment:
Should there be an explicit assertion here? All this really shows is that it doesn't explode if you call it twice.